org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/parser/AbstractCommentParser.java

```diff
@@ -180,9 +180,13 @@ protected boolean commentParse() {
 			int lastStarPosition = -1;
 
 			// Init scanner position
+			this.markdown = this.source[this.javadocStart + 1] == '/';
 			this.linePtr = getLineNumber(this.firstTagPosition);
-			int realStart = this.linePtr==1 ? this.javadocStart : this.scanner.getLineEnd(this.linePtr-1)+1;
-			if (realStart < this.javadocStart) realStart = this.javadocStart;
+			int realStart = this.javadocStart;
+			if (!this.markdown) {
+				realStart = this.linePtr==1 ? this.javadocStart : this.scanner.getLineEnd(this.linePtr-1)+1;
+				if (realStart < this.javadocStart) realStart = this.javadocStart;
```

Comment:
I wonder if the difference is really between markdown and traditional, or perhaps `firstTagPosition` is the culprit, which may point to a `@` that actually does not start a tag because it's inside a code block. 
But avoiding a bogus firstTagPosition would require even the scanner (where `commentTagStarts` is maintained) to detect code blocks, which doesn't sound like a task for the scanner.

Do you know how this is handled in traditional javadoc, i.e., does the scanner signal a tag when we have `{@code @param is not a tag here}`?

Reply:
> I see that even with a regular Javadoc, the firstTagPosition does point to the @code. So, there's no difference there. Just that the parser handles it after that.

That **is** a difference: in traditional javadoc you *need* a tag even to enter a code section, in markdown looking for the first `@` is not sufficient to understand the relevant structure.

IOW, starting at `{@code` is fine, everything before that is uninteresting for semantic analysis. But in markdown starting at `@param` lets you miss that fact that we are inside a code block.